src/Volume.ts

```diff
@@ -229,14 +235,28 @@ export default class Volume {
     this.normRegionOffset = subregionOffset.clone().divide(volumeSize);
   }
 
-  updateRequiredData(required: Partial<LoadSpec>, onChannelLoaded?: PerChannelCallback) {
+  /** Call on any state update that may require new data to be loaded (subregion, enabled channels, time, etc.) */
+  async updateRequiredData(required: Partial<LoadSpec>, onChannelLoaded?: PerChannelCallback): Promise<void> {
     this.loadSpecRequired = { ...this.loadSpecRequired, ...required };
+    let noReload =
+      this.loadSpec.time === this.loadSpecRequired.time &&
+      this.loadSpec.subregion.containsBox(this.loadSpecRequired.subregion);
+
+    // An update to `subregion` should trigger a reload when the new subregion is not contained in the old one
+    // OR when the new subregion is smaller than the old one by enough that we can load a higher scale level.
+    if (noReload && !this.loadSpec.subregion.equals(this.loadSpecRequired.subregion)) {
+      const currentScale = this.imageInfo.multiscaleLevel;
+      // `LoadSpec.multiscaleLevel`, if specified, forces a cap on the scale level we can load.
+      const minScale = this.loadSpec.multiscaleLevel ?? 0;
+      // Loaders should cache loaded dimensions so that this call blocks no more than once per valid `LoadSpec`.
+      const dims = await this.loader?.loadDims(this.loadSpecRequired);
+      if (dims && currentScale > minScale && dims[currentScale - 1].canLoad) {
```

Comment:
I wonder if the canLoad could be a dynamic check here?  I don't think it's too slow to call pickLevelToLoad if it doesn't require network fetches but I forget how frequently this gets called.

Reply:
I guess I felt like the decision of which level to load was the loader's responsibility, since `pickLevelToLoad` and the `MAX_ATLAS_DIMENSION` constant live in the same file as `OMEZarrLoader`. I kind of agree though that this ought to be treated more like a global limit. Do we make that official by moving those items into VolumeLoaderUtils and importing/calling `pickLevelToLoad` here? This is as opposed to calling it internally in `loadDims`, which is what is done now.

`pickLevelToLoad` itself doesn't require network fetches, it just takes dimensions and performs a bit of logic. Those dimensions, however, need to be fetched. This is the problem I referred to in the PR description and in the comment on line 251 above: loaders need to be smart enough to ensure that providing dimensions rarely requires a fetch in practice. The behavior in this PR is:
- In `OMEZarrLoader`, all dims are fetched before loader construction and `loadDims` will never block
- In `TiffLoader`, metadata is fetched on the first call to `loadDims` or `loadVolumeData` and cached (so if the volume already has data, `loadVolumeData` has been called and `loadDims` won't block in practice)
- In `JsonImageInfoLoader`, metadata is fetched on the first call to `loadDims` or `loadVolumeData` *per timestep* and cached. In practice, if this function calls `loadVolumeData` after calling `loadDims`, the first `loadDims` call will have cached the metadata that the `loadVolumeData` would have required so checking dims first adds no additional work.

Moving the `pickLevelToLoad` call won't change these rules on when this function has to wait for the network.

Reply:
Yea I basically agree with everything you said here.  
* pickleveltoload can be a utils-like shared function, or even a IVolumeLoader method that returns 0 by default and only does something nontrivial in omezarrloader.
* all dims could be fetched at or close to loader creation time.  Then they will just be available and no call to loadDims would have to fetch anything (though they could if they wanted?)